python/feathub/processors/processor_job.py

```diff
@@ -24,6 +24,16 @@ class ProcessorJob(ABC):
     def __init__(self) -> None:
         pass
 
+    @abstractmethod
+    def cancel(self) -> None:
```

Comment:
Would it useful to return a `Future` similar to `pyflink.common.JobClient.cancel()`?

Reply:
After looking at the `pyflink.common.JobClient`, I think the Future is only used for acknowledgment. It is completed as soon as the server confirms the cancel is received but not when the job is canceled. 

Currently, the cancel method implies the same semantics as the method should return after the cancel is acknowledged. We may need to state that in the docstring explicitly. Given that the two ways have the same semantics and we are currently not using `CompletableFuture` in other places, I am leaning toward the current way. What do you think?

Reply:
After the offline discussion, I agree it is better to return the Future object and users can decide if they want to wait for acknowledgment or not.